core/capabilities/examples/on_demand_trigger.go

```diff
@@ -0,0 +1,77 @@
+package examples
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities"
+)
+
+var info = capabilities.MustCapabilityInfo(
+	"on-demand-trigger",
+	capabilities.CapabilityTypeTrigger,
+	"An example on-demand trigger.",
+	"v1.0.0",
+)
+
+// OnDemandTrigger is an example on-demand trigger.
+type OnDemandTrigger struct {
+	capabilities.CapabilityInfo
+	chans map[string]chan capabilities.CapabilityResponse
+	mu    sync.Mutex
+}
+
+// NewOnDemandTrigger returns a new OnDemandTrigger.
+func NewOnDemandTrigger() *OnDemandTrigger {
+	return &OnDemandTrigger{
+		CapabilityInfo: info,
+		chans:          map[string]chan capabilities.CapabilityResponse{},
+	}
+}
+
+func (o *OnDemandTrigger) FanOutEvent(ctx context.Context, event capabilities.CapabilityResponse) error {
+	o.mu.Lock()
+	defer o.mu.Unlock()
+	for _, ch := range o.chans {
+		ch <- event
+	}
+	return nil
+}
+
+// Execute executes the on-demand trigger.
+func (o *OnDemandTrigger) SendEvent(ctx context.Context, wid string, event capabilities.CapabilityResponse) error {
```

Comment:
I'm wondering if this is a good example. Should a trigger ever support targeting a particular workflow ID? It seems like listeners should be transparent to triggers. Let me know if I'm missing an obvious use case.